### DISCUSSION TITLE
Access to typesetting info

### DISCUSSION THREAD

**Discussion by IRobot1** [protectwise/troika#239] [Ideas]
Currently, troika-three doesn't support scrolling the text within a fixed area.  If the lines calculated by the typesetting could be exposed, it should be possible to pass in a block of text, get back the array of lines that would be rendered.  Instead of rendering all the lines, I could pass in just the lines that would fit in the area.  Scrolling would involve just changing the index of the first line to be rendered.

Imagine I passed in the following text.  

```
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
```

Based on the width, the typesetter might break it into the following lines of text

```
Lorem Ipsum is simply dummy text 
of the printing and typesetting industry. 
Lorem Ipsum has been the industry's 
standard dummy text ever since the 1500s,
when an unknown printer took a galley of 
type and scrambled it to make a type 
specimen book.
```

However, I want to limit the height of the text being rendered to only the first 3 lines.  Now that I know how the typesetter will render the lines, I can just pass in the text for the first three lines.  

```
Lorem Ipsum is simply dummy text 
of the printing and typesetting industry. 
Lorem Ipsum has been the industry's 
```

To scroll, I would change the index of the first of the three lines

```
of the printing and typesetting industry. 
Lorem Ipsum has been the industry's 
standard dummy text ever since the 1500s,
```

Does this sound reasonable?  Can this capability be added?

**Comment by lojjic** [comment]
Scrolling is typically implemented by using a shifting `clipRect` and `position.y`. Here's a [sandbox demonstrating that approach](https://codesandbox.io/s/troika-three-text-scrolling-9nhyz9?file=/src/App.js) using R3F. Does that meet your needs?
